Frontend/LoginState.cs

```diff
@@ -5,24 +5,25 @@
 namespace Frontend;
 
 public class LoginState {
-    public bool Fetched { get; private set; }
+
+    private readonly HttpClient _httpClient;
 
     [MemberNotNullWhen(true, nameof(User))]
     public bool LoggedIn => User != null;
 
     public User? User { get; private set; }
+    
+    public Task Initialization { get; private set; }
 
     public LoginState(HttpClient client) {
-#pragma warning disable CS4014
-        Check(client);
-#pragma warning restore CS4014
+        _httpClient = client;
+        Initialization = CheckAsync();
     }
 
-    private async Task Check(HttpClient client) {
-        var cnt = await client.GetAsync("auth");
+    private async Task CheckAsync() {
+        var cnt = await _httpClient.GetAsync("auth");
         if (cnt.IsSuccessStatusCode) {
-            User = await cnt.Content.ReadFromJsonAsync<User>();
+           User = await cnt.Content.ReadFromJsonAsync<User>();
         }
-        Fetched = true;
```

Comment:
In case this sticks it needs to use a private constructor with an async factory method.

Reply:
However, at that point we can just revert to cascading a user since this class provides no additional functionality

Reply:
Unless you want to consider `Task.IsCompleted` as the alternative to `Fetched` in which case that changes nothing but lower the level of abstraction.

Reply:
I would have used `await LoginState.Initialization` in such a scenario. Yes, compared to a cascading user this doesn't provide much. I'm also not sure you'd need to call the constructor the way it is called right now. Feels like this could be solved better.

Reply:
Nvm doing so would probably trigger an unnecessary re-render, `Task.IsCompleted` might be better.